src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -469,16 +479,14 @@ impl LogicalAggBuilder {
         agg_call: AggCall,
     ) -> std::result::Result<ExprImpl, ErrorCode> {
         let return_type = agg_call.return_type();
-        let (agg_kind, inputs, distinct, mut order_by, filter) = agg_call.decompose();
+        let (agg_kind, inputs, mut distinct, mut order_by, filter) = agg_call.decompose();
         match &agg_kind {
-            AggKind::Min
-            | AggKind::Max
-            | AggKind::Sum
-            | AggKind::Count
-            | AggKind::Avg
-            | AggKind::ApproxCountDistinct => {
-                // this order by is unnecessary.
-                order_by = OrderBy::new(vec![]);
+            AggKind::Min | AggKind::Max => {
+                distinct = false;
```

Comment:
Why is distinct always set to false here 🤔

Reply:
Duplicate items won't affect the result of `min`/`max`

Reply:
Wait actually it will right? If min max is stateful. Then we receive a delete.
If there are duplicate for distinct agg wouldn't it contain extra of the same item, and may not update the min/max correctly?

Reply:
Duplicate items won't affect the result of min/max, so `min(distinct a)` equals to `min(a)`, right? As long as `min` aggregater can handle duplicate items correctly, then the rewriting is correct. And ExtremeAgg we have now can indeed handle duplicate items correctly, it sorts all input rows in its state table.

Reply:
Nevermind I think I got confused. Even if we receive delete, it will still revert to duplicated value.

Reply:
> Duplicate items won't affect the result of min/max, so `min(distinct a)` equals to `min(a)`, right? As long as `min` aggregater can handle duplicate items correctly, then the rewriting is correct. And ExtremeAgg we have now can indeed handle duplicate items correctly, it sorts all input rows in its state table.

Thanks for clarifying 👍